From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Timo Alho <talho@nvidia.com>
Date: Wed, 15 May 2024 10:08:29 +0300
Subject: [PATCH 006/254] NVIDIA: SAUCE: clk: tegra: make clock names unique
 for each die

Each clock name that is registered to the clock framework has to have
a unique name.
In order to achieve that when registering clocks in a multidie setup,
append the die number as suffix to the clock names.

Signed-off-by: Timo Alho <talho@nvidia.com>
Change-Id: Ie471165aecb30a5d70ebba0565df84d84a1a4c7b
(cherry picked from commit 670400430ae7fb2e630b75c64783f60fd5eff653)
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3180934
Tested-by: Rakesh Babu Bodla <rbodla@nvidia.com>
Reviewed-by: Rakesh Babu Bodla <rbodla@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>
(cherry picked from commit 33ad608d36571d07d7da254e2ffecd5477c36c54)

diff --git a/drivers/clk/tegra/clk-bpmp.c b/drivers/clk/tegra/clk-bpmp.c
index 77a2586dbe000..cbc8579c6cce7 100644
--- a/drivers/clk/tegra/clk-bpmp.c
+++ b/drivers/clk/tegra/clk-bpmp.c
@@ -19,7 +19,7 @@
 
 struct tegra_bpmp_clk_info {
 	unsigned int id;
-	char name[MRQ_CLK_NAME_MAXLEN];
+	char name[MRQ_CLK_NAME_MAXLEN + 2];
 	unsigned int parents[MRQ_CLK_MAX_PARENTS];
 	unsigned int num_parents;
 	unsigned long flags;
@@ -367,7 +367,12 @@ static int tegra_bpmp_clk_get_info(struct tegra_bpmp *bpmp, unsigned int id,
 	if (err < 0)
 		return err;
 
-	strscpy(info->name, response.name, MRQ_CLK_NAME_MAXLEN);
+	if (dev_to_node(bpmp->dev) == NUMA_NO_NODE)
+		strscpy(info->name, response.name, MRQ_CLK_NAME_MAXLEN);
+	else
+		snprintf(info->name, sizeof(info->name), "%s.%d",
+			 response.name, dev_to_node(bpmp->dev));
+
 	info->num_parents = response.num_parents;
 
 	for (i = 0; i < info->num_parents; i++)
-- 
2.43.0

